maestro-ios-driver/src/main/kotlin/xcuitest/XCTestDriverClient.kt

```diff
@@ -197,16 +197,22 @@ class XCTestDriverClient(
         return okHttpClient.newCall(requestBuilder.build()).execute()
     }
 
-    private fun OkHttpClient.Builder.addRetryOnErrorInterceptor() = addInterceptor(Interceptor {
-        val request = it.request()
-        try {
-            it.proceed(request)
-        } catch (connectException: IOException) {
-            installer.start()?.let { newClient ->
-                client = newClient
-                it.proceed(request)
-            } ?: throw XCTestDriverUnreachable("Failed to reach out XCUITest Server in RetryOnError")
+    private fun OkHttpClient.Builder.addRetryInterceptor() = addInterceptor(Interceptor { chain ->
+        var exception: IOException? = null
+        repeat(3) {
```

Comment:
Didn't we a have a retrying mechanism already in place? How this is different?

Reply:
Also there is https://square.github.io/okhttp/4.x/okhttp/okhttp3/-ok-http-client/-builder/retry-on-connection-failure/
API if we want to retry on connection failures.

Reply:
The current retrying mechanism does not retry based on the status code of the response. So it only retries on connection failure (through an exception). Now this retry is more generic will retry based if the status code is not successful (200)

Reply:
> Now this retry is more generic will retry based if the status code is not successful (200)

But why do we need that having can 4xx response be flaky from server?